Tests/src/PublicTestBase.h

```diff
@@ -17,8 +17,8 @@
 
 #include <gtest/gtest.h>
 
-
-constexpr char* EndpointBaseURI = "https://ogs-internal.magnopus-dev.cloud";
+constexpr char* EndpointEnvironmentName = "MAGNOPUS_SERVICES_ENDPOINT";
+extern const char* EndpointBaseURI;
```

Comment:
[Consider] I wonder if this is vulnerable to the static-initialization-order-fiasco. Would we be more comfortable with something like
```suggestion
const char* GetEndpointBaseURI() {
  return GetEnvironmentVariableOrDefault(EndpointEnvironmentName, "https://ogs-internal.magnopus-dev.cloud");
}
```

Reply:
Yeah this would be a lot nicer in general, however was attempting to keep the changes limited and have as few files as possible be effect in the change.

Reply:
As for static-initialization-order this shouldn't be a problem due to the use of extern deferring the definition and directly defined in the cpp file. But of course would be good to check.

Reply:
Granted currently, we are likely safe by coincidence, this sort of approach is still vulnerable.

Take this example : https://onlinegdb.com/VzZqSFk82

In that example, two global variables are defined in this manner, one depending on the other. Depending on the order of compilation (undefined), the variable that depends on the prior one may be garbage, (is is during my execution.)

It's a pretty bad idea to leave these sorts of footguns in place when we can avoid them.

[Cpp core standards section.](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#i22-avoid-complex-initialization-of-global-objects)

Reply:
should be resolved as part of recent commits, went with static initialization ensure we don't run into these pit falls, as you mentioned it would be best resolved these changes before they become a potential problem down the line

Reply:
Thanks for doing that.